### PR TITLE
smmstore tool

### PR DESCRIPTION
**Pull request by PLangowski**

### CODE REVIEW

**Comment by macpijan** [comment]
@PLangowski We have a long-forgotten simple DTS  I have introduced in the beginning: https://github.com/Dasharo/meta-dts/blob/main/meta-dts-distro/recipes-tests/dts-tests/dts-tests/cukinia.conf
It should not hurt us too much to add one or two extra lines with each MR here to add some basic testing and maintain the DTS quality just a little bit higher

File: meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+DESCRIPTION = "Offline SMMSTORE variable modification too"
```
Comment (macpijan) [inline meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb:1]:
```suggestion
DESCRIPTION = "Offline SMMSTORE variable modification tool"
```

File: meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+LICENSE = "GPL-2.0-or-later"
+
+SRC_URI = " \
+git://github.com/coreboot/coreboot.git;branch=main;protocol=https \
```
Comment (macpijan) [inline meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb:5]: isn't main the default branch anyway?

File: meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+file://a5df001.diff;patchdir=${WORKDIR}/git \
```
Comment (macpijan) [inline meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb:6]: I prefer adding at least 4 spaces here when we break lines. This is rather common if you look at other recipes, to have some indentation here.

File: meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+inherit pkgconfig
+
+S = "${WORKDIR}/git/util/smmstoretool"
+
+do_install() {
+    oe_runmake 'DESTDIR=${D}' install
+}
+
+FILES:${PN} += " \
```
Comment (macpijan) [inline meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb:23]: Why `+=` ?

File: meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+   /usr/local/bin/smmstoretool \
```
Comment (macpijan) [inline meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb:24]: Why not ${bindir} ?

**Review by macpijan — CHANGES_REQUESTED** [review]

File: meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb (hunk 2 above)
Comment (PLangowski) [reply meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb:5]: I think BitBake searches `master` by default. `coreboot` has `main` instead, so the fetcher fails if the branch is not provided
```
ERROR: smmstoretool-git-r0 do_fetch: Fetcher failure: Unable to find revision 602653abed391ae1b1445ad86d0f05b8b5b678cb in branch master even from upstream
```

File: meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb (hunk 5 above)
Comment (PLangowski) [reply meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb:24]: `${bindir}` expands to `/usr/bin`, but the [Makefile ](https://github.com/coreboot/coreboot/blob/main/util/smmstoretool/Makefile#L11) installs the tool in `/usr/local/bin` by default. Is that location okay or should I change it?

File: meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb (hunk 1 above)
Comment (PLangowski) [reply meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb]: Fixed 308b71460416f4da9cfb6027e82cf9654487a592

File: meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb (hunk 3 above)
Comment (PLangowski) [reply meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb]: My bad; fixed 308b71460416f4da9cfb6027e82cf9654487a592

File: meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb (hunk 4 above)
Comment (PLangowski) [reply meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb]: 308b71460416f4da9cfb6027e82cf9654487a592

File: meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb (hunk 5 above)
Comment (macpijan) [reply meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb:24]: You should ovveride the PREFIX variable in Yocto recipe to adhere to typical paths in Yocto.

**Comment by PLangowski** [comment]
I added a cukinia check 3e46f3eccc87

```
bash-5.2# cukinia
[PASS] Check whether the ectool package can be started
[PASS] Check whether the ifdtool package can be started
[PASS] Check whether the intelp2m package can be started
[PASS] Check whether the inteltool package can be started
[PASS] Check whether the msrtool package can be started
[PASS] Check whether the nvramtool package can be started
[PASS] Check whether the superiotools package can be started
[PASS] Check whether the dmidecode package can be started
[PASS] Check whether the iasl package can be started
[PASS] Check if flashrom supports Tiger Lake
[PASS] Check if flashrom supports ite_ec programmer
[PASS] Check if dasharo-hcl-report script exists
[PASS] Check is smmstoretool package can be started
result: 0 failure(s)
```

File: meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb (hunk 5 above)
Comment (PLangowski) [reply meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb]: Is this what you mean? 33cda5107591077ab6db0d141c5a911f29130b2d

Comment (macpijan) [reply meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb]: Definitely not. Please take a look at other recipes (e.g. cbmem), even in meta-dts. It is very common pattern.

Comment (PLangowski) [reply meta-dts-distro/recipes-bsp/txe-secure-boot/smmstoretool_git.bb]: Done f26221db11a05bcbede4bc06596a5983acb02373

**Pull request merged**